internal/subject/database/repository.go

```diff
@@ -1,84 +1,116 @@
 package database
 
 import (
+	"fmt"
+
 	"github.com/google/uuid"
 	"github.com/jmoiron/sqlx"
 
 	"github.com/sumelms/microservice-course/internal/subject/domain"
 	"github.com/sumelms/microservice-course/pkg/errors"
 )
 
-type Repository struct {
-	*sqlx.DB
+const (
+	listSubject    = "list subject"
+	getSubject     = "get subject by uuid"
+	delecteSubject = "delete subject by uuid"
+)
+
+func NewRepository(db *sqlx.DB) (repository, error) {
+	sqlStatements := make(map[string]*sqlx.Stmt)
+
+	for queryName, query := range queries() {
+		stmt, err := db.Preparex(string(query))
+		if err != nil {
+			return repository{}, errors.WrapErrorf(err, errors.ErrCodeUnknown, "error preparing statement %s", queryName)
+		}
+		sqlStatements[queryName] = stmt
+	}
+	fmt.Println(sqlStatements)
+	return repository{
+		statements: sqlStatements,
+	}, nil
+}
+
+type repository struct {
+	statements map[string]*sqlx.Stmt
 }
 
-func (r *Repository) Subject(id uuid.UUID) (domain.Subject, error) {
-	query := `SELECT * FROM subjects WHERE uuid = $1`
+type Query string
```

Comment:
I'll move line 39-47 under the consts definition at the top of the file, or (maybe better) into a separated file named `queries.go` at the same level of the `database/repository.go` file.

Reply:
This is nice because when we will mock the database we will be able to reuse the queries if we have it as const. So I will move it.